lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -701,6 +701,47 @@ sub added_entities_counts {
     return \%result;
 }
 
+sub secondary_counts {
+    my ($self, $editor_id, $viewing_own_profile) = @_;
+
+    my $editor = $self->get_by_id($editor_id);
+    $self->load_preferences($editor);
+
+    my %result;
+
+    if ($viewing_own_profile || $editor->preferences->public_tags) {
+        my $upvote_count = 0;
+        my $downvote_count = 0;
+        for my $type (entities_with('tags')) {
+            my $query = "SELECT is_upvote, COUNT(*) FROM ${type}_tag_raw WHERE editor = ? GROUP BY is_upvote";
+            my $rows = $self->sql->select_list_of_lists($query, $editor_id);
+
+            for my $row (@$rows) {
+                my ($is_upvote, $count) = @$row;
+                if ($is_upvote) {
+                    $upvote_count += $count;
+                } else {
+                    $downvote_count += $count;
+                }
+            }
+        }
```

Comment:
I think we should be able to reduce the number of round-trips to the server by doing this in one query. Maybe something like:

```SQL
SELECT x.is_upvote, count(*) FROM (
  SELECT is_upvote FROM work_tag_raw WHERE editor = ?
  UNION ALL
  SELECT is_upvote FROM recording_tag_raw where editor = ?
  UNION ALL
  ...
) x
GROUP BY x.is_upvote;
```

etc. though you'd have to build the inner part of the query of course.

Reply:
My only worry about this was to hardcode the tables, but maybe that's not a problem.

Reply:
Oh, actually, I guess I can do something similar to https://github.com/metabrainz/musicbrainz-server/blob/49f0a33667130378141f081ad4511284aa95358c/lib/MusicBrainz/Server/Data/Editor.pm#L477

Reply:
That seemed to work :)